types/convert.go

```diff
@@ -73,8 +73,12 @@ func IntergerSignedUpperBound(intType byte) int64 {
 		return math.MaxInt32
 	case mysql.TypeLonglong:
 		return math.MaxInt64
+	case mysql.TypeEnum:
+		// enum can have at most 65535 distinct elements
+		// it would be better to use len(FieldType.GetElems()), but we only have a byte type here
+		return 65535
```

Comment:
It is MaxUint64 in IntergerUnsignedUpperBound. Besides, do we need to cover IntergerSignedLowerBound?

Reply:
Nice catch. I also modify it to 65536. I think 65536 is a more reasonable value.

> Besides, do we need to cover IntergerSignedLowerBound?

It would be better to set it to zero :thinking:, though I don't know whether it's reachable.